src/main/java/dal/fatih/todoly/HibernateTaskRepository.java

```diff
@@ -0,0 +1,83 @@
+package dal.fatih.todoly;
+
+
+import javax.persistence.*;
+import java.sql.Date;
+import java.util.*;
+
+public class HibernateTaskRepository implements TaskRepository {
+
+    private final EntityManagerFactory entityManagerFactory
+            = Persistence.createEntityManagerFactory("Todoly");
+    private final EntityManager entityManager
+            = entityManagerFactory.createEntityManager();
+    private final TypedQuery<Task> listQuery
+            = entityManager.createQuery("select a from Task a ", Task.class);
+    private final TypedQuery<Task> getQuery
+            = entityManager.createQuery("select a from Task a where a.taskId = ?1", Task.class);
+    private final Query deleteQuery
+            = entityManager.createQuery("delete from Task where taskId = ?1");
+    private final Query filterQuery
+            = entityManager.createNativeQuery("select * from Task where dueDate  " +
+            "between now() and ?1", Task.class);
+    private final Query getByTitleOrDesQuery
+            = entityManager.createNativeQuery("select * from Task where title like ?1 " +
+            "or description like ?2", Task.class);

```

Comment:
The app itself is pretty small. No need for using prepared statements. It was good for you to learn them but move these to their respective methods. Like list query to the listing method, create query o the creating method and so on. 

This means no more `?1`, `?2` etc.